### PR TITLE
Populates timeline with reactions information 

### LINKED ISSUE #50: Populate timeline with likes and comments count, and if it is liked by the logged user

### PR DESCRIPTION
**Pull request by itumejia**
(isLiked, likes and comments counts)

Kapture 2021-07-30 at 20 29 29

### CODE REVIEW

File: app/src/main/java/com/example/dogbook/main/fragments/TimelineFragment.java [hunk 1]
```diff
@@ -78,23 +81,44 @@ private void setUpRecyclerView() {
...
         });
+
+//        ParseQuery<Post> query = ParseApplication.getAllPostsQuery();
+//        query.findInBackground(new FindCallback<Post>() {
+//            @Override
+//            public void done(List<Post> objects, ParseException e) {
+//                if (e == null) {
+//                    posts.clear();
+//                    posts.addAll(objects);
+//                    postsAdapter.notifyDataSetChanged();
```
Comment (itumejia) [inline app/src/main/java/com/example/dogbook/main/fragments/TimelineFragment.java:112]: Forgot to delete comments, will do in next PR

Comment (nicholeic) [reply app/src/main/java/com/example/dogbook/main/fragments/TimelineFragment.java:112]: Let's not merge next time if there's code that's commented out

File: app/src/main/java/com/example/dogbook/main/models/Post.java [hunk 2]
```diff
@@ -23,12 +27,27 @@
...
     private int noComments;
 
     public Post() {}
 
+    //Add reactions information to the posts
+    public static List<Post> addReactions(List<Post> posts, List<HashMap> reactions) {
+        for (int i = 0; i < posts.size(); i++) {
+            posts.get(i).setLiked((Boolean) reactions.get(i).get(KEY_REACTIONS_IS_LIKED));
+            posts.get(i).setNoLikes((int) reactions.get(i).get(KEY_REACTIONS_LIKES_COUNT));
+            posts.get(i).setNoComments((int) reactions.get(i).get(KEY_REACTIONS_COMMENTS_COUNT));
```
Comment (nicholeic) [inline app/src/main/java/com/example/dogbook/main/models/Post.java:45]: Commented on the prior diff I believe -- naming could be better here

File: backend code_cloud_functions/main.js [hunk 3]
```diff
@@ -6,44 +6,47 @@ const CLASS_LIKE = "Like";
...
-        object["commentsCount"] = getCommentCount(object.id);
-        timeline.push(object);
+        object["isLiked"] = await isPostLiked(posts[i], request.user);
+        object["likesCount"] = await getLikesCount(posts[i]);
+        object["commentsCount"] = await getCommentCount(posts[i]);
+        reactions.push(object);
     }
 
-    return timeline;
+    return {posts, reactions};
```
Comment (nicholeic) [inline backend code_cloud_functions/main.js:30]: You could consider returning a wrapper model that is a pair of:
```
List<Pair<Post, Reactions>>
```

Then, you wouldn't need to worry about keeping those lists ordered but instead would have 1 model object that contains both.  Or you could even add a variable/field to the "object"/dictionary have right now that is the "post"

File: backend code_cloud_functions/main.js [hunk 4]
```diff
@@ -6,44 +6,47 @@ const CLASS_LIKE = "Like";
...
+    query.descending(KEY_CREATED_AT);
+    query.include(KEY_AUTHOR);
 
     const posts = await query.find();
-    let timeline = []
+    let reactions = []
 
     for (let i = 0; i < posts.length; i++) {
-        let object = posts[i];
+        let object = {};
```
Comment (nicholeic) [inline backend code_cloud_functions/main.js:22]: Let's improve this name.  What does the object represent?

**Pull request merged**